Edit wallet name

nop33 opened the pull request:
Closes #132

nop33 commented on src/App.tsx:
```diff
@@ -251,10 +251,10 @@ const Main = ({ children, ...props }: ViewProps) => {
   useEffect(() => {
     const handleAppStateChange = (nextAppState: AppStateStatus) => {
-      if (nextAppState === 'background' && !isCameraOpen) {
+      if (nextAppState === 'background' && walletMnemonic && !isCameraOpen) {
```
We only trigger the "appBecameInactive" when we actually have a mnemonic (meaning, a wallet). Otherwise, it means we don't have one yet, so we are in the wallet creation flow.

nop33 commented on src/App.tsx:
```diff
@@ -251,10 +251,10 @@ const Main = ({ children, ...props }: ViewProps) => {
         navigateRootStack('SplashScreen')
         dispatch(appBecameInactive())
-      } else if (nextAppState === 'active' && !walletMnemonic && !isUnlockingApp && !isCameraOpen) {
+      } else if (nextAppState === 'active' && !walletMnemonic && !isCameraOpen && !isUnlockingApp) {
```
Just a reordering to match the order of the condition above 😇

nop33 commented on src/components/AuthenticationModal.tsx:
```diff
@@ -48,22 +49,28 @@ const errorInstructionSet: Instruction[] = [
     try {
-      const storedWallet = await getStoredWallet(usePin)
-      const usesBiometrics = usePin ? false : await loadBiometricsSettings()
+      const storedWallet = await getStoredWallet(forcePinUsage)
+
+      // This should never happen, but if it does, inform the user instead of being stuck
+      if (!storedWallet) {
+        Alert.alert('Missing wallet', 'Could not find wallet to authenticate. Please, restart the app')
+        return
+      }
```
This component should only be used when we have a wallet and we want to authenticate. So, in the event that this is used without a wallet (which should be never), we should still catch it and make sure the user doesn't get stuck.

nop33 commented on src/components/AuthenticationModal.tsx:
```diff
@@ -113,12 +117,16 @@ const ConfirmWithAuthModal = ({ onConfirm, onClose, usePin = false }: ConfirmWit
           <PinCodeInput pinLength={pinLength} onPinEntered={decryptMnemonic} />
         </ModalContent>
+      ) : (
+        <Message>
+          <AppText>Loading wallet...</AppText>
+        </Message>
```
To make sure there is never just a white screen. Should only be displayed while the fingerprint sensor is visible.

nop33 commented on src/components/ModalWithBackdrop.tsx:
```diff
@@ -19,13 +19,13 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
 
 const ModalWithBackdrop = ({ children, closeModal, color, ...props }: ModalWithBackdropProps) => (
-  <Modal transparent={true} {...props} animationType="none">
+  <Modal transparent={true} animationType="none" {...props}>
```
so that `animationType` can be overridden if needed.

mvaivre replied:
Yeah I forced it to none because these animations posed problems in the past. Let's just keep that in mind in case we see witness behaviours in the future :)

nop33 replied:
It's good to default to none, but be able to override it if needed though, don't you agree?

nop33 replied:
We are currently using `"fade"` in these places:
- Spinner modal
- Consolidation modal
- Color picker modal
- Verify Mnemonic modal
- Settings screen

We should check each individual case on iOS to make sure we don't get stuck anywhere.

nop33 commented on src/screens/LoginWithPinScreen.tsx:
```diff
@@ -58,9 +60,20 @@ const LoginWithPinScreen = (props: LoginWithPinScreenProps) => {
     <Screen contrastedBg {...props}>
-      {isPinModalVisible && <ConfirmWithAuthModal usePin onConfirm={handleSuccessfulLogin} />}
+      <AuthenticationModal visible={isPinModalVisible} forcePinUsage onConfirm={handleSuccessfulLogin} />
+      {!isPinModalVisible && (
+        <Message>
+          <AppText>Nice 👍 Unlocking...</AppText>
+        </Message>
+      )}
```
This will be shown while we are loading some initial data when the app is first launched. Then it will disappear much quicker after data has been initialized.

mvaivre replied:
What about showing the loading spinner instead? Just a bit less confusing IMO as you don't have to read anything, you understand that you just need to wait.

nop33 replied:
Sure. My initial thought is that by having some text it'd be easier to debug in case someone gets stuck on this screen for whatever reason. I'll leave the word "Unlocking" with a spinner, if you don't mind.

nop33 replied:
Done ✅

mvaivre reviewed (commented):
🚂 👏

nop33 commented on src/screens/AddressDiscoveryScreen.tsx:
```diff
@@ -67,6 +68,9 @@ const AddressDiscoveryScreen = ({ navigation, route: { params }, ...props }: Scr
   const stopScan = () => dispatch(addressDiscoveryStopped())
 
+  const continueToNextScreen = () =>
+    resetNavigationState(isImporting ? 'NewWalletSuccessScreen' : 'InWalletTabsNavigation')
```
Resetting the navigation state helps with preventing going back to the wallet creation procedure. There's no point doing that since the wallet has already been created at this point.

nop33 commented:
I rebased it on `internal-feedback-polish-2` (after all the work we did today with the `/android` and `/ios` folders) and also added my "custom network" changes so that we only have 1 PR open each.

Tomorrow morning we can start by merging yours, then reviewing, testing and merging this one.

@mvaivre, I would focus the functional tests on your iPhone device and mainly to try all functions that show a spinner modal: https://github.com/alephium/mobile-wallet/pull/297#discussion_r1349969208

nop33 commented on src/components/AddressCard.tsx:
```diff
@@ -124,8 +124,9 @@ const AddressCard = ({ style, addressHash, onSettingsPress }: AddressCardProps)
           shadowRadius: 8,
-          elevation: 10,
-          borderColor: outterBorderColor
+          elevation: 3,
+          borderColor: outterBorderColor,
+          overflow: 'hidden'
```
I had to change the elevation and also add the overflow, cause it didn't look as good on Android as it did on iOS:

<img width="489" alt="image" src="https://github.com/alephium/mobile-wallet/assets/1579899/0afcc98d-1294-45f1-b439-5bf9de503a93">
